meta-genivi-dev/meta-ivi/recipes-extended/node-state-manager/node-state-manager_2.0.0.bbappend

```diff
@@ -1,5 +1,21 @@
-EXTRA_OECONF_append=" --with-nsmc=NodeStateMachineTest "
+#
+# This bbappend will add the Simple Node State Machine to the
+# node-state-manager build.
+#
+NSMC="SimpleNodeStateMachine"
 
-do_install_append() {
-    install -m 755 ${B}/NodeStateMachineTest/NodeStateTest ${D}${bindir}/NodeStateTest
-}
+EXTRA_OECONF_append=" --with-nsmc=${NSMC} "
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+NSMC_REV="b4de5dee4d70a223eef37bcbb5f8536560b0e243"
+
+#
+# The SRC_URI append will checkout the commit at the location
+# hard-coded/configured in the EXTRA_OECONF above.
+# Also patch the node-state-manager configure.ac to build the
+# simple-node-state-machine.
+#
+SRC_URI_append = " \
+    git://github.com/GENIVI/simple-node-state-machine.git;protocol=https;rev=${NSMC_REV};destsuffix=git/${NSMC} \
+    file://0001-Build-${NSMC}.patch \
+"
```

Comment:
Please refer https://github.com/openembedded/meta-openembedded/blob/master/meta-oe/recipes-support/opencv/opencv_3.2.bb.

Use ;name and SRCREV_name ...